exporter/zipkin/lib/opentelemetry/exporter/zipkin/transformer.rb

```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+require 'ipaddr'
+
+module OpenTelemetry
+  module Exporter
+    module Zipkin
+      # @api private
+      module Transformer
+        extend self
+
+        # based on https://github.com/openzipkin/zipkin-ruby/blob/master/lib/zipkin-tracer/trace.rb
+        # https://github.com/openzipkin/zipkin-go/blob/0b3ebdbc2ddf7409f84316407fec22faf1ce8a0f/model/kind.go#L21-L26https://github.com/openzipkin/zipkin-go/blob/0b3ebdbc2ddf7409f84316407fec22faf1ce8a0f/model/kind.go#L21-L26
+        # https://github.com/open-telemetry/opentelemetry-collector/blob/347cfa9ab21d47240128c58c9bafcc0014bc729d/translator/trace/zipkin/traces_to_zipkinv2.go#L269-L280# based on https://github.com/open-telemetry/opentelemetry-collector/blob/347cfa9ab21d47240128c58c9bafcc0014bc729d/translator/trace/zipkin/traces_to_zipkinv2.go#L269-L280
+        KIND_MAP = {
+          OpenTelemetry::Trace::SpanKind::SERVER => 'SERVER',
+          OpenTelemetry::Trace::SpanKind::CLIENT => 'CLIENT',
+          OpenTelemetry::Trace::SpanKind::PRODUCER => 'PRODUCER',
+          OpenTelemetry::Trace::SpanKind::CONSUMER => 'CONSUMER',
+          OpenTelemetry::Trace::SpanKind::INTERNAL => nil
+        }.freeze
+
+        SERVICE_NAME_ATTRIBUTE_KEY = 'service.name'
+        ERROR_TAG_KEY = 'error'
+        STATUS_CODE_NAME = 'otel.status_code'
+        STATUS_UNSET = 'UNSET'
+        STATUS_ERROR = 'ERROR'
+        STATUS_OK = 'OK'
+        ATTRIBUTE_PEER_SERVICE = 'peer.service'
+        ATTRIBUTE_NET_PEER_IP = 'peer.ip'
+        ATTRIBUTE_NET_PEER_PORT = 'peer.port'
+        ATTRIBUTE_NET_HOST_IP = 'host.ip'
+        ATTRIBUTE_NET_HOST_PORT = 'host.port'
+
+        DEFAULT_SERVICE_NAME = OpenTelemetry::SDK::Resources::Resource.default.attribute_enumerator.find { |k, _| k == SERVICE_NAME_ATTRIBUTE_KEY }&.last || 'unknown_service'
+        private_constant(:KIND_MAP, :DEFAULT_SERVICE_NAME, :SERVICE_NAME_ATTRIBUTE_KEY, :ERROR_TAG_KEY, :STATUS_CODE_NAME, :STATUS_UNSET, :STATUS_ERROR, :STATUS_OK)
+
+        def to_zipkin_span(span_d, resource)
+          start_time = (span_d.start_timestamp.to_f * 1_000_000).to_i
+          duration = (span_d.end_timestamp.to_f * 1_000_000).to_i - start_time
+          tags = {}
+          service_name = DEFAULT_SERVICE_NAME
+          resource&.attribute_enumerator&.select do |key, value|
```

Comment:
```suggestion
          resource.attribute_enumerator.select do |key, value|
```
`resource` and `resource.attribute_enumerator` should always be non-nil.